pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -93,17 +92,16 @@ class NativeAssetsBuildRunner {
   /// performed without linking (i.e. [linkingEnabled] is `false`).
   ///
   /// The native assets build runner does not support reentrancy for identical
-  /// [BuildConfig] and [LinkConfig]! For more info see:
+  /// [BuildInput] and [LinkInput]! For more info see:
   /// https://github.com/dart-lang/native/issues/1319
   Future<BuildResult?> build({
-    required BuildConfigCreator configCreator,
-    required BuildConfigValidator configValidator,
+    required BuildInputCreator inputCreator,
+    required BuildInputValidator inputValidator,
     required BuildValidator buildValidator,
     required ApplicationAssetValidator applicationAssetValidator,
     required Uri workingDirectory,
     PackageLayout? packageLayout,
     String? runPackageName,
-    required List<String> buildAssetTypes,
```

Comment:
Why did you remove this?

The current code has separation based on how the protocol is layered:

* The optional things / protocol extensions are handled by embedder by calling out to it (e.g. code config)
* The core protocol is handled in `package:native_assets_builder`, this field is core protocol support

So the `build` and `link` methods should take all things it needs for the core protocol, thereby ensuring those are setup as required by core protocol.

By moving it to the embedder, the `package:native_assets_builder` can no longer guarantee the core protocol fields are setup, it relies on the embedder to handle core protocol functionality - which doesn't seem right to me.

Reply:
Oops, I forgot to revert this when I added `buildAssetTypes` back in. Thanks!